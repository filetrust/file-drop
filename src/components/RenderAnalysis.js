import React from "react";
import Items from "./Items";

function RenderAnalysis(props) {
  return (
    <div className="analysis-results">
      <div className="sanitisationTable table-container">
        <h1 className="table-header">
          Active content that has been sanitised (removed)
        </h1>
        <table>
          <tbody>
            <Items items={props.sanitisations} />
          </tbody>
        </table>
      </div>
      <br />
      <div className="remediationsTable table-container">
        <h1 className="table-header">
          Objects & Structures that have been repaired
        </h1>
        <table>
          <tbody>
            <Items items={props.remediations} />
          </tbody>
        </table>
      </div>
      <br />
      <div className="issuesTable table-container">
        <h1 className="table-header">
          Objects & Structures that are unable to be repaired
        </h1>
        <table>
          <tbody>
            <Items items={props.issues} />
          </tbody>
        </table>
      </div>
    </div>
  );
}

export default RenderAnalysis;
