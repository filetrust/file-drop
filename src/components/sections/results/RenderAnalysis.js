import React from "react";
import Items from "./Items";

function RenderAnalysis(props) {
  return (
    <div className="analysis-results">
      <div className="sanitisationTable analysis-table">
        <div className="h1 table-header">
          Active content that has been sanitised (removed)
        </div>
        <table>
          <tbody>
            <Items items={props.sanitisations} />
          </tbody>
        </table>
      </div>

      <div className="remediationsTable analysis-table">
        <div className="h1 table-header">
          Objects & Structures that have been repaired
        </div>
        <table>
          <tbody>
            <Items items={props.remediations} />
          </tbody>
        </table>
      </div>

      <div className="issuesTable analysis-table">
        <div className="h1 table-header">
          Objects & Structures that are unable to be repaired
        </div>
        <table>
          <tbody>
            <Items items={props.issues} />
          </tbody>
        </table>
      </div>
    </div>
  );
}

export default RenderAnalysis;
