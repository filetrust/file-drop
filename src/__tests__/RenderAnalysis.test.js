import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RenderAnalysis from "../components/RenderAnalysis";

Enzyme.configure({adapter: new Adapter()});

describe("RenderAnalysis", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<RenderAnalysis />);
    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<RenderAnalysis />);
    expect(wrapper).toMatchSnapshot();
  });
});
