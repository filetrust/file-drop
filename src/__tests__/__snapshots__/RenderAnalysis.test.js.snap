// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderAnalysis matches snapshot 1`] = `ShallowWrapper {}`;
